java.base/src/main/java/java/net/NetworkInterface$_qbicc.java

```diff
@@ -0,0 +1,537 @@
+        boolean isVirtual = false;
+
+        // If the interface name is a logical interface then we remove the unit
+        // number so that we have the physical interface (eg: hme0:1 -> hme0).
+        // NetworkInterface currently doesn't have any concept of physical vs.
+        // logical interfaces.
+        strncpy(addr_of(name[0]), if_name, IF_NAMESIZE.cast());
+        name[IF_NAMESIZE.intValue() - 1] = word(0);
+        vname[0] = word(0);
+
+        // Create and populate the netaddr node.
+        netaddr addrP = new netaddr();
+        size_t addr_size = (family == AF_INET) ? sizeof(struct_sockaddr_in.class) : sizeof(struct_sockaddr_in6.class);
+        addrP.addr = malloc(addr_size);
+        if (addrP.addr.isNull()) {
+            return ifs; // match OpenJDK behavior: if malloc fails just return the unmodified netif
+        }
+        memcpy(addrP.addr.cast(), ifr_addrP.cast(), addr_size);
+        addrP.family = family;
+        addrP.mask = prefix;
+        addrP.next = null;
+
+        // for IPv4 add broadcast address
+        if (family == AF_INET && !ifr_broadaddrP.isNull()) {
+            addrP.brdcast = malloc(addr_size);
+            if (addrP.brdcast.isNull()) {
+                return ifs; // match OpenJDK behavior: if malloc fails just return the unmodified netif
+            }
+            memcpy(addrP.brdcast.cast(), ifr_broadaddrP.cast(), addr_size);
+        } else {
+            addrP.brdcast = word(0);
+        }
+
+        // Deal with virtual interface with colon notation e.g. eth0:1
+        int colonIdx = 0;
+        for (int i=0; i<IF_NAMESIZE.intValue(); i++) {
+            if (name[i].intValue() == word(':').intValue()) {
+                colonIdx = i;
+                break;
+            }
+        }
+        if (colonIdx != 0) {
+            c_int flags = auto(word(0));
+            // This is a virtual interface. If we are able to access the parent
+            // we need to create a new entry if it doesn't exist yet *and* update
+            // the 'parent' interface with the new records.
+            name[colonIdx] = word('\0');
+            if (getFlags(sock, addr_of(name[0]), addr_of(flags)).intValue() < 0 || flags.intValue() < 0) {
+                // failed to access parent interface do not create parent.
+                // We are a virtual interface with no parent.
+                isVirtual = true;
+                name[colonIdx] = word(':');
+            } else {
+                // Got access to parent, so create it if necessary.
+                // Save original name to vname and truncate name by ':'
+                memcpy(addr_of(vname[0]).cast(), addr_of(name[0]).cast(), IF_NAMESIZE.cast());
+                vname[colonIdx] = word(':');
+            }
+        }
+
+        // Check if this is a "new" interface. Use the interface name for
+        // matching because index isn't supported on Solaris 2.6 & 7.
+        while (currif != null) {
```

Comment:
Since `currif` is not a pointer, I sense this would fail. At least, it should. Auto storage is meant to mimic the behavior of C when it allocates a variable on the stack - you can take its address, but the variable itself is treated like a value.

Reply:
Same here.  currif is an instance of the Java `netif` class, not a C struct